### PR TITLE
Move changelog to docs

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR moves the changelog (currently `CHANGES.rst` in the root of the project repository) to the sphinx docs. Also makes use of the sphinx `extlinks` extension to link to related issues and pull requests in the changelog easily.

### CODE REVIEW

**Comment by jrbourbeau** [comment]
I wasn't sure which `toctree` in `index.rst` the changelog should go in. Couldn't decide between the "Getting started with repo2docker" section or the "Contribute to repo2docker" section, so I made a new "Changelog" `toctree`. Happy to modify as needed.

File: docs/source/conf.py [hunk 1]
```diff
@@ -33,7 +33,12 @@
 # Add any Sphinx extension module names here, as strings. They can be
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
-extensions = []
+extensions = ['sphinx.ext.extlinks']
+
+extlinks = {
+    'issue': ('https://github.com/jupyter/repo2docker/issues/%s', 'Issue #'),
+    'pr': ('https://github.com/jupyter/repo2docker/pull/%s', 'PR #')
```
[the remark this answers is not in the document]
Comment (jrbourbeau) [reply docs/source/conf.py]: Yeah, for sure! Good idea

**Comment by betatim** [comment]
Bimbambosh! Thanks again. I learnt about `extlinks`!

**Comment by jrbourbeau** [comment]
Thanks for reviewing again! Yeah, `extlinks` is super useful

For future reference, where can I find the rendered preview of the docs? I saw they get saved as a CircleCI artifact, I'm just not sure where to get the corresponding link to view them (e.g. you posted https://135-90722372-gh.circle-artifacts.com/0/html/index.html earlier)

**Comment by betatim** [comment]
If you go to the CircleCI page can you see the "Artifacts" tab (kinda in the center of the screenshot)?

screen shot 2018-12-17 at 16 41 28

**Comment by betatim** [comment]
My guess is that only people who are maintainers or the like can see them. Makes me wonder if there is a GitHub bot already that can copy a link to the index page into a comment on a PR that modified the docs.

Maybe we can modify the code for https://github.com/apps/notebookbot that does something like this for notebooks? If you think that would be remotely useful create an issue or leave a thumbs up here and I'll create the issue.

**Comment by jrbourbeau** [comment]
I think modifying/extending notebookbot to post documentation build links would be great!

**Pull request merged**